solver/llbsolver/util/util.go

```diff
@@ -8,21 +8,25 @@ import (
 	controlapi "github.com/moby/buildkit/api/services/control"
 )
 
-// DedupCacheOptions removes duplicates from cacheOpts
-func DedupCacheOptions(cacheOpts []*controlapi.CacheOptionsEntry) (result []*controlapi.CacheOptionsEntry, err error) {
+func FindDuplicateCacheOptions(cacheOpts []*controlapi.CacheOptionsEntry) ([]*controlapi.CacheOptionsEntry, error) {
 	seen := map[string]*controlapi.CacheOptionsEntry{}
+	duplicate := map[string]struct{}{}
```

Comment:
My suggestion was to return the list of types as the caller ends up needing it anyways:
```go
func FindDuplicateCacheOptions(cacheOpts []*controlapi.CacheOptionsEntry) (result []string, err error) {
	seen := map[string]struct{}{}
	duplicates := map[string]struct{}{}
	for _, opt := range cacheOpts {
		k, err := cacheOptKey(*opt)
		if err != nil {
			return nil, err
		}
		if _, ok := seen[k]; !ok {
			seen[k] = struct{}{}
		} else {
			duplicates[opt.Type] = struct{}{}
		}
	}
	result = make([]string, 0, len(duplicates))
	for k := range duplicates {
		result = append(result, k)
	}
	return result, nil
}
```

Reply:
I think the function is better generic, and not specific to the exact message that the caller prints out later.